runtime/doc/api.txt

```diff
@@ -3363,6 +3363,13 @@ nvim_create_autocmd({event}, {*opts})                  *nvim_create_autocmd()*
         })
 <
 
+    Example using a User autocommand: >lua
+        vim.api.nvim_create_autocmd("User", {
+          pattern = "MyPlugin",
+          command = "echo 'got MyPlugin event'",
```

Comment:
This may add slightly more confusion because it's not a MyPlugin event, it's a User event.

```suggestion
          command = "echomsg 'got User event'",
```

This and https://github.com/neovim/neovim/pull/24793 are special-cases because of usability issues with the User event.

Reply:
Maybe there could be both in the string for the sake of explicitness? i.e. `"echomsg 'got MyPlugin user event"`

While updating the User event docs would probably help, I think that because most autocommands have a file name glob as their pattern, there's an impulse to assume that all patterns are file name globs, and that something special is going on when encountering the User event which does something else.

Reply:
Yes, but that's an unfounded assumption. There's two things at play here:
1. technical documentation, which needs to be clear and correct but not "robust against prior miscomprehension";
2. user manual, which _should_ guard against such miscomprehension.

It is a difficulty that (Neo)Vim's documentation is sprawling and contains both. I agree with @justinmk that the best place for a better explanation is the canonical place, which is `:h User`. For example, we can add a (single) sentence such as (addition bold):


> Not executed automatically.  Use |:doautocmd| to trigger this, typically for "custom events"			in a plugin.  **Specify a custom pattern to distinguish different events or plugins.** Example:  



>  "echomsg 'got MyPlugin user event"

This is not great, as it again conflates the intention ("event triggered by MyPlugin") from the implementation ("`User` event with `MyPlugin` pattern").